pkgs/games/fteqw/default.nix

```diff
@@ -0,0 +1,98 @@
+{ lib
+, stdenv
+, pkgs
+, fetchsvn
+, runCommand
+, pkgName ? "fteqw"
+}:
+let
+  releaseFile = {
+    fteqw = pkgName;
+    fteqw-dedicated = "fteqw-sv";
+    fteqcc = pkgName;
+  };
+in stdenv.mkDerivation rec {
+  pname = pkgName;
+  version = "unstable-2022-08-09";
+  src = fetchsvn {
+    url = "https://svn.code.sf.net/p/fteqw/code/trunk";
+    rev = "6303";
+    sha256 = "sha256-tSTFX59iVUvndPRdREayKpkQ+YCYKCMQe2PXZfnTgPQ=";
+  };
+  makeFlags = [
+    "PKGCONFIG=pkg-config"
+    "-C engine"
+  ];
+  buildFlags = lib.optionals (pkgName == "fteqw") [
+    "m-rel"
+  ] ++ lib.optionals (pkgName == "fteqw-dedicated") [
+    "sv-rel"
+  ] ++ lib.optionals (pkgName == "fteqcc") [
+    "qcc-rel"
+  ];
+  buildInputs = with pkgs; lib.optionals (pkgName == "fteqw") [
+      gzip libvorbis libmad SDL2 SDL2_mixer
+      libpng alsa-lib gnutls libjpeg vulkan-loader
+      speex libopus xorg.libXrandr xorg.libXcursor
+  ] ++ lib.optionals (pkgName == "fteqw-dedicated") [
+      gnutls zlib
+  ] ++ lib.optionals (pkgName == "fteqcc") [
+      zlib
+  ];
+  nativeBuildInputs = with pkgs; lib.optionals (pkgName == "fteqw") [
+      SDL2.dev pkg-config libpng.dev gnutls.dev
+      xorg.libXrandr.dev vulkan-loader.dev vulkan-headers
+  ] ++ lib.optionals (pkgName == "fteqw-dedicated") [
+      gnutls.dev pkg-config zlib.dev
+  ] ++ lib.optionals (pkgName == "fteqcc") [
+      zlib.dev pkg-config
+  ];
+  enableParallelBuilding = true;
+  preBuild = ''
+      substituteInPlace ./engine/Makefile \
+        --replace "I/usr/include/opus" "I${pkgs.libopus.dev}/include/opus"
+      substituteInPlace ./engine/gl/gl_vidlinuxglx.c \
+        --replace 'Sys_LoadLibrary("libXrandr"' 'Sys_LoadLibrary("${pkgs.xorg.libXrandr}/lib/libXrandr.so"'
+    '';
```

Comment:
This either goes into postPatch when the fils already exist in the source tarball or into postConfigure when they only exist after configurePhase.